lib/import-export-visitor.js

```diff
@@ -303,15 +318,16 @@ IEVp.finalizeHoisting = function () {
     delete bodyInfo.hoistedExportsMap;
     delete bodyInfo.hoistedExportsString;
     delete bodyInfo.hoistedImportsString;
-  }, this);
+  }
 
   // Just in case we call finalizeHoisting again, don't hoist anything.
   this.bodyInfos.length = 0;
 
-  this.removals.forEach(processRemoval);
+  let removalCount = this.removals.length;
 
-  // Just in case we call finalizeHoisting again, don't remove anything.
-  this.removals.length = 0;
+  while (removalCount--) {
+    processRemoval(this.removals.pop());
```

Comment:
I'm not a fan of micro-optimizations but with everything there's context. In general use I'd say `Array#forEach`, `bind`, and all the built-ins are great. But for something specialized like a game engine or a parser micro-opts can have a cumulative impact. The things I'm suggesting aren't as brittle as other micro-opts. These are things that have held up for 12+ years so less likely to change from release to release of the engine.